### DISCUSSION TITLE
Don't understand this code

### DISCUSSION THREAD

**Discussion by Zo-Bro-23** [anuraghazra/github-readme-stats#2407] [Q&A] [answered]
https://github.com/anuraghazra/github-readme-stats/blob/0ff426d258387ecd0ebbb9206d91b96781e61cf0/src/cards/stats-card.js#L295-L300

I don't understand this code. First of all, why are we using "Total commits" label and adding `totalStars` value? Second of all, why do we even need a special case for commits? `STATS.commits.label` and `STATS.commits.value` will work right?

**Comment by Zo-Bro-23** [comment, marked as answer]
@rickstaa any idea why this is?

**Reply by rickstaa** [reply]
Of course. It adds the year behind `Total Commits` when `include_all_commits` is disabled.

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=anuraghazra)](https://github.com/anuraghazra/github-readme-stats)

**Reply by Zo-Bro-23** [reply]
But `STATS[key].label` already includes the label with the year added.

https://github.com/anuraghazra/github-readme-stats/blob/0ff426d258387ecd0ebbb9206d91b96781e61cf0/src/cards/stats-card.js#L137-L139

I don't see the need to have separate logic for commits when the general logic will work just as well.

**Reply by rickstaa** [reply]
If you have more code questions, feel free to join my discord server (see my [README.md](https://github.com/rickstaa)). Talking there is a bit easier and @anuraghazra is also in there. 👍🏻

**Reply by rickstaa** [reply]
> But `STATS[key].label` already includes the label with the year added.
> 
> https://github.com/anuraghazra/github-readme-stats/blob/0ff426d258387ecd0ebbb9206d91b96781e61cf0/src/cards/stats-card.js#L137-L139
> 
> I don't see the need to have separate logic for commits when the general logic will work just as well.

Last time I checked it did not 🤔.

https://github.com/anuraghazra/github-readme-stats/blob/8bc69e761ff7c922f47e6561d7a2f4fd0cc153c7/src/translations.js#L75-L104

**Reply by Zo-Bro-23** [reply]
> If you have more code questions, feel free to join my discord server (see my [README.md](https://github.com/rickstaa)). Talking there is a bit easier. 👍🏻

Do you mind sticking to GitHub, please? I find it easier to reference code, issues, and PRs here than on Discord. Markdown is also a ***lot*** easier for formatting and structuring that plain text 😊

**Reply by Zo-Bro-23** [reply]
> Last time I checked it did not :thinking:

So there's nothing wrong with this code, per se. It just does the same thing twice. I've removed this in #2409 though, so it'll be fixed when that gets merged.

**Reply by rickstaa** [reply]
> > If you have more code questions, feel free to join my discord server (see my [README.md](https://github.com/rickstaa)). Talking there is a bit easier. 👍🏻
> 
> Do you mind sticking to GitHub, please? I find it easier to reference code, issues, and PRs here than on Discord. Markdown is also a _**lot**_ easier for formatting and structuring that plain text 😊

I also prefer using GitHub for project-related stuff.👍 The discord was to explain to you some things about the maintenance of the project (e.g. the way @anuraghazra and I are planning to move this project into the future), which are quite lengthy and do not consider users that are not a collaborator 😅.

**Reply by rickstaa** [reply]
> > Last time I checked it did not 🤔
> 
> So there's nothing wrong with this code, per se. It just does the same thing twice. I've removed this in #2409 though, so it'll be fixed when that gets merged.

Can you link to me where you think the year is already added? 🤔

**Reply by rickstaa** [reply]
BTW @anuraghazra I think already send you a collaboration invite. 🚀

**Reply by Zo-Bro-23** [reply]
> Can you link to me where you think the year is already added? :thinking:

https://github.com/anuraghazra/github-readme-stats/blob/0ff426d258387ecd0ebbb9206d91b96781e61cf0/src/cards/stats-card.js#L128-L142

**Reply by Zo-Bro-23** [reply]
> BTW @anuraghazra I think already send you a collaboration invite. :rocket:

Didn't get an email, but maybe it'll take time!

**Reply by rickstaa** [reply]
> > Can you link to me where you think the year is already added? 🤔
> 
> https://github.com/anuraghazra/github-readme-stats/blob/0ff426d258387ecd0ebbb9206d91b96781e61cf0/src/cards/stats-card.js#L128-L142

Good catch. One of those can be removed.👍
